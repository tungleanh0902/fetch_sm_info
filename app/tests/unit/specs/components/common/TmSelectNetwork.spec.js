import { shallowMount } from "@vue/test-utils"
import TmSelectNetwork from "common/TmSelectNetwork"

describe(`TmSelectNetwork`, () => {
  let wrapper, $store

  const networks = [
    {
      id: `awesomenet`,
      testnet: true,
      default: false,
    },
    {
      id: `keine-ahnungnet`,
      testnet: true,
      default: false,
    },
    {
      id: `la-red-feliz`,
      testnet: false,
      default: true,
    },
    {
      id: `localnet`,
      testnet: true,
      default: false,
    },
    {
      id: `emilys-chain`,
      testnet: false,
      default: false,
    },
  ]

  beforeEach(async () => {
    $store = {
      dispatch: jest.fn(),
      getters: {
        network: `localnet`,
        networks,
      },
    }
    wrapper = shallowMount(TmSelectNetwork, {
      mocks: {
        $router: {
          push: jest.fn(),
        },
        $route: {
          params: {},
        },
        $store,
      },
    })
    self.networkId = `localnet`
  })

  it(`has the expected html structure`, async () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
