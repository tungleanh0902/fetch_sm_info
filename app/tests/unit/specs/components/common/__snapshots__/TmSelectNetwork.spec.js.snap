// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TmSelectNetwork has the expected html structure 1`] = `
<sessionframe-stub
  icon=""
>
  <div
    class="select-network"
  >
    <h2
      class="session-title"
    >
      Select a Network
    </h2>
     
    <networklist-stub
      networks="[object Object],[object Object]"
      sectiontitle="Main Networks"
    />
     
    <networklist-stub
      networks="[object Object],[object Object],[object Object]"
      sectiontitle="Test Networks"
    />
  </div>
</sessionframe-stub>
`;
